fleet/postgres/agents.go

```diff
@@ -30,6 +30,41 @@ type agentRepository struct {
 	logger *zap.Logger
 }
 
+func (r agentRepository) RetrieveMatchingAgents(ctx context.Context, tags types.Tags) (types.Metadata, error) {
+	t, tmq, err := getOrbTagsQuery(tags)
+	if err != nil {
+		return types.Metadata{}, errors.Wrap(errors.ErrSelectEntity, err)
+	}
+
+	q := fmt.Sprintf(
+		`select
+			json_build_object('total', coalesce(total,0), 'online', coalesce(online,0)) AS matching_agents
+		from
+			(select
+				sum(case when mf_thing_id is not null then 1 else 0 end) as total,
+				sum(case when state = 'online' then 1 else 0 end) as online
+			from agents WHERE mf_thing_id is not null %s) as agent_groups`, tmq)
```

Comment:
security problem: this query does not limit for OwnerID so it will query agents from all tenants